release notes/v0.35.0.md

```diff
@@ -0,0 +1,94 @@
+
+### JS promises are now an option 
+
+With the goja update in [#2197](https://github.com/grafana/k6/pull/2197), you can now make a `Promise` and chaining in your k6 scripts:
+
+```js
+export default function () {
+    var p = new Promise((resolve, reject) => {
+        console.log('do something promising!');
+        reject('here');
+    });
+
+    p.then(
+        (s) => { console.log('fulfilled with', s) },
+        (s) => { console.log('rejected with', s) },
+    );
+}
+```
+
+### InfluxDB v2 API
+
+After hard work at working out how to integrate InfluxDB v2 API, it was decided to pull that integration into an [xk6-output-influxdb extension](https://github.com/grafana/xk6-output-influxdb). InfluxDB v1 API is still available as before, with minor optimizations ([#2190](https://github.com/grafana/k6/pull/2190)).
+
+### New xk6 extensions
+
+With this release, we have added two brand-new xk6 extensions:
+
+- [xk6-browser](https://github.com/grafana/xk6-browser) is a browser automation extension which relies on [Chrome Devtools Protocol](https://chromedevtools.github.io/devtools-protocol/).
+- [xk6-output-remote-write](https://github.com/grafana/xk6-output-prometheus-remote) is an output extension with support for [Prometheus Remote-Write protocol](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit#heading=h.3p42p5s8n0ui).
+
+### Other notable changes
+
+- k6's GRPC was extended with a support for server reflection which allows one to use GRPC even without `proto` file at hand. You can read more about the protocol [here](https://github.com/grpc/grpc/blob/master/doc/server-reflection.md) ([#2160](https://github.com/grafana/k6/pull/2160)).
+- Support for cookies in `k6/ws` ([#2193](https://github.com/grafana/k6/pull/2193)).
+- Forbid `metric.Add` to let `NaN` values through ([#1876](https://github.com/grafana/k6/pull/1876)); instead, print out nice warnings or throw an exception if `--throw` is enabled. These warnings will not become too large ([#2219](https://github.com/grafana/k6/pull/2219)).
+- Support for compression in websockets ([#2162](https://github.com/grafana/k6/pull/2162)). Thanks, @cooliscool!
+
+## Breaking changes
+
+- Addition of common metrics registry no longer allows defining custom metrics with the same name as one of the builtin metrics, e.g. `new Counter("http_req_duration")` will abort. Builtin metrics may no longer be referenced as global objects in xk6 extensions either ([#1832](https://github.com/grafana/k6/issues/1832)).
+- Fix inconsistency in environment variables' names: use `K6_NO_SETUP` and `K6_NO_TEARDOWN` options instead of `NO_SETUP` and `NO_TEARDOWN` ([#2140](https://github.com/grafana/k6/pull/2140)).
+
+## Bugs fixed!
+
+- Fix of a misleading sorting of custom submetrics ([#2198](https://github.com/grafana/k6/pull/2198)). Thanks, @knittl!
+- Fix for `afero.FS` bug which caused some `afero`-dependent xk6 extensions to fail ([#2216](https://github.com/grafana/k6/pull/2216)).
```

Comment:
```suggestion
- Implement a newer version of the `afero.FS` interface for internal filesystems, so that extension depending on that or newer version can be built ([#2216](https://github.com/grafana/k6/pull/2216)).
```